plugin/core/protocol.py

```diff
@@ -720,5298 +813,5150 @@ class TokenFormat(StrEnum):
-    'textDocument': 'TextDocumentIdentifier',
-    # The position inside the text document.
-    'position': 'Position',
-    # An optional token that a server can use to report work done progress.
-    'workDoneToken': NotRequired['ProgressToken'],
-    # An optional token that a server can use to report partial results (e.g. streaming) to
-    # the client.
-    'partialResultToken': NotRequired['ProgressToken'],
-})
+class DeclarationParams(TypedDict):
+    textDocument: 'TextDocumentIdentifier'
+    """ The text document. """
+    position: 'Position'
+    """ The position inside the text document. """
+    workDoneToken: NotRequired['ProgressToken']
+    """ An optional token that a server can use to report work done progress. """
+    partialResultToken: NotRequired['ProgressToken']
+    """ An optional token that a server can use to report partial results (e.g. streaming) to
+    the client. """
 
 
-TypeDefinitionRegistrationOptions = TypedDict('TypeDefinitionRegistrationOptions', {
-    # A document selector to identify the scope of the registration. If set to null
-    # the document selector provided on the client side will be used.
-    'documentSelector': Union['DocumentSelector', None],
-    # The id used to register the request. The id can be used to deregister
-    # the request again. See also Registration#id.
-    'id': NotRequired[str],
-})
+class DeclarationRegistrationOptions(TypedDict):
+    documentSelector: Union['DocumentSelector', None]
+    """ A document selector to identify the scope of the registration. If set to null
+    the document selector provided on the client side will be used. """
+    id: NotRequired[str]
+    """ The id used to register the request. The id can be used to deregister
+    the request again. See also Registration#id. """
 
 
-WorkspaceFolder = TypedDict('WorkspaceFolder', {
-    # The associated URI for this workspace folder.
-    'uri': 'URI',
-    # The name of the workspace folder. Used to refer to this
-    # workspace folder in the user interface.
-    'name': str,
-})
-""" A workspace folder inside a client. """
+class SelectionRangeParams(TypedDict):
+    """ A parameter literal used in selection range requests. """
+    textDocument: 'TextDocumentIdentifier'
+    """ The text document. """
+    positions: List['Position']
+    """ The positions inside the text document. """
+    workDoneToken: NotRequired['ProgressToken']
+    """ An optional token that a server can use to report work done progress. """
+    partialResultToken: NotRequired['ProgressToken']
+    """ An optional token that a server can use to report partial results (e.g. streaming) to
+    the client. """
 
 
-DidChangeWorkspaceFoldersParams = TypedDict('DidChangeWorkspaceFoldersParams', {
-    # The actual workspace folder change event.
-    'event': 'WorkspaceFoldersChangeEvent',
-})
-""" The parameters of a `workspace/didChangeWorkspaceFolders` notification. """
+class SelectionRange(TypedDict):
```

Comment:
Is using `Class` equivalent to the old way in terms of overhead (memory, speed)?

Reply:
One noticable reason on why I want to switch to `class SelectionRange(TypedDict):` form `SelectionRange = TypedDict('SelectionRange', {` syntax is because classes: allows for doc comments:
<img width="1058" alt="Screenshot 2024-07-02 at 12 46 32" src="https://github.com/sublimelsp/LSP/assets/22029477/3e224246-1fc5-4a25-be61-e016a343d2d8">

The previous syntax that we used was an alternative syntax for TypedDict for python version less than 3.5.

I haven't done any benchmarks. I didn't notice any noticeable improvements or slowness.